x-pack/plugins/security_solution/public/management/services/policies/hooks.ts

```diff
@@ -83,3 +84,27 @@ export function useGetEndpointSecurityPackage({
     customQueryOptions
   );
 }
+
+export function useBulkGetAgentPolicies({
+  isEnabled,
+  policyIds,
+}: {
+  isEnabled: boolean;
+  policyIds: string[];
+}): QueryObserverResult<BulkGetAgentPoliciesResponse['items'], IHttpFetchError> {
+  const http = useHttp();
+
+  return useQuery<BulkGetAgentPoliciesResponse['items'], IHttpFetchError>(
+    ['agentPolicies', policyIds],
+
+    async () => {
+      if (policyIds.length > 0) {
+        return (await sendBulkGetAgentPolicies({ http, requestBody: { ids: policyIds } }))?.items;
```

Comment:
I’m not a huge fan of this approach. It feels like we’re overemphasizing a side effect - the error thrown by useQuery that causes this value to be undefined. In this case, I’m referring to a scenario where the API returns an error. Additionally, I believe the check for undefined isn’t being executed here since the error occurs during the fetch and gets intercepted by useQuery. Let me know your thoughts.

Reply:
yeah, i guess you're right, this is really unnecessary, it's not executed on error

Reply:
39aa138eaa1de26f5b0b09ed086534930402fa2f

Reply:
For the specific check with the optional chaining `?.` in return (`await sendBulkGetAgentPolicies({ http, requestBody: { ids: policyIds } }))?.items;`), I believe it won’t be executed if `sendBulkGetAgentPolicies` throws an error. While the functionality will still work due to useQuery catching the error and returning an Error object, the side effect is that `referencedAgentPolicies` will end up being undefined.

It might be worth considering if it makes sense to handle this fetch and potential errors more explicitly.

Reply:
you are right, `referencedAgentPolicies` will be `undefined` in case of an error - but, as its type is `AgentPolicy[] | undefined` by definition (coming from `useQuery`), the code which uses the variable must handle `undefined`. and it does: in case of an error we fallback to the agent policy ID instead of its name:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/5bc2c3e6-cfad-49f8-85db-b521ee322229">

so i would say this should be okay for now. the only issue is that the error message is hidden from the user, but everything should work without a successful fetching, so at least we're more fault tolerant and show less annoying error messages :)